File: flask_backend/routes/public.py
from flask import Blueprint, request, jsonify
from database.db import get_db

public_bp = Blueprint('public', __name__)

@public_bp.route('/profile/<string:token>', methods=['GET'])
def get_public_profile(token):
    try:
        db = get_db()
        emp = next((e for e in db['employees'] if e.get('secure_token') == token), None)
        if not emp:
            return jsonify({"success": False, "message": "Profile not found"}), 404
            
        data = {
            "name": emp.get('name'),
            "designation": emp.get('designation'),
            "department": emp.get('department'),
            "dob": emp.get('dob'),
            "gender": emp.get('gender'),
            "phone": emp.get('phone'),
            "emergency_contact": emp.get('emergency_contact'),
        }
        return jsonify({"success": True, "data": data})
    except Exception as e:
        print(e)
        return jsonify({"success": False, "message": "Failed to load profile"}), 500

File: backend/routes/public.js
const express = require('express');
const router = express.Router();
const { getDB } = require('../database/db');

// ─── GET /api/public/profile/:token ───────────────────────────────────────────
router.get('/profile/:token', (req, res) => {
  try {
    const db = getDB();
    const emp = db.prepare('SELECT * FROM employees WHERE secure_token = ?').get(req.params.token);
    if (!emp) return res.status(404).json({ success: false, message: 'Profile not found' });

    res.json({
      success: true,
      data: {
        name: emp.name,
        designation: emp.designation,
        department: emp.department,
        dob: emp.dob,
        gender: emp.gender,
        phone: emp.phone,
        emergency_contact: emp.emergency_contact,
      },
    });
  } catch (err) {
    res.status(500).json({ success: false, message: 'Failed to load profile' });
  }
});

module.exports = router;
